### PR TITLE
Customizable protocol

### PR DESCRIPTION
**Pull request by 38elements**
In the current implementation, the developer can not change the behavior of the protocol.
By implementing a subclass of HttpProtocol and making it an argument of the `run` function, 
the developer can change the behavior when data is received and the behavior when an error occurs.
This is related #115 and #208 .

### CODE REVIEW

File: sanic/server.py [hunk 1]
```diff
@@ -221,12 +221,13 @@ def trigger_events(events, loop):
                 loop.run_until_complete(result)
 
 
-def serve(host, port, request_handler, error_handler, before_start=None,
-          after_start=None, before_stop=None, after_stop=None,
-          debug=False, request_timeout=60, sock=None,
+def serve(protocol, host, port, request_handler, error_handler,
```
Comment (seemethere) [inline sanic/server.py]: Would it be smart to put this at the front of the arguments list and possibly break functionality for anyone who uses this? I think putting this at the end and defaulting it to our own HttpProtocol would be better in this case.

**Comment by 38elements** [comment]
@seemethere 
Thank you for reviewing.
Your pointing out is correct.
I fixed this code.

**Comment by 38elements** [comment]
@seemethere 
Of course I will make document and test.

**Comment by 38elements** [comment]
@seemethere 
I made document, example and test.

**Pull request merged**